### DISCUSSION TITLE
Exit segment no longer working?

### DISCUSSION THREAD

**Discussion by kamfaima** [JanDeDobbeleer/oh-my-posh#3490] [Troubleshoot] [answered]
Here's my exit segment in my custom OMP theme file which I'm using only in VS Code 1.75.1 on macOS 13.1. The problem seems to happen ONLY in VS Code using the PowerShell Integrated Terminal. If I run it in a standard pwsh terminal then it works as expected.

```
{
          "background": "#2e9599",
          "background_templates": [
            "{{ if gt .Code 0 }}#f1184c{{ end }}"
          ],
          "foreground": "#ffffff",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "properties": {
            "always_enabled": false
          },
          "style": "diamond",
          "trailing_diamond": "\ue0b4",
          "type": "exit"
}
```

The logic is based on the default(?) exit segment but I've modified it to only show an error status if the previous command returned an error status. (I believe this is done by setting `always_enabled` property to `false`.) See the following screenshot of when it is working.

![image](https://user-images.githubusercontent.com/23546392/218449629-14490c7e-dfc2-45e9-8e99-a4d4af1858cb.png)

However the last few weeks, this has stopped working and the segment never shows an error. I put in this line of code           `"template": "Code: {{ .Code }} | Meaning: {{ .Meaning }}"` and changed `always_enabled` property to `true` to see what the error code is and it's always 0.

Here it is NOT working:

![image](https://user-images.githubusercontent.com/23546392/218449436-dd1be57f-72cc-468c-bdb2-fb6d20210ed7.png)

So I guess the question is why does it work in a "standard" pwsh terminal but not in VS Code's PowerShell extension terminal?

Lastly, how do I get VS Code to respect the text colour (white, like in the first screenshot) as per the theme config? I've noticed that the text only turns white when I use a very dark background. Changing the VS Code theme to a light theme doesn't seem to have any effect.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@kamfaima you probably have shell integrations enabled for vscode which is what influences this negatively. That's a bug on their end, and known if I'm not mistaken. About the color, that's a bug in vscode's terminal.

**Reply by kamfaima** [reply]
Ok thanks for leading me to that. For now I have disabled VS Code's shell integrations by setting `"terminal.integrated.shellIntegration.enabled": false` and the exit segment is working again.
